docs/csharp/tutorials/exploration/csharp-7.yml

```diff
@@ -0,0 +1,536 @@
+
+    Now that the method returns a reference to the integer value in the
+    matrix, you need to modify where it's called.  The `var` declaration
+    means that `valItem` is now an `int` rather than a tuple:
+
+    [!code-csharp[AssignRefReturnToValue](~/samples/snippets/csharp/new-in-7/program.cs#23_AssignRefReturnToValue "Assign ref return to value")]
+
+    The second `WriteLine` statement in the example above prints out the value `42`,
+    not `24`. The variable `valItem` is an `int`, not a `ref int`. The `var`
+    keyword enables the compiler to specify the type, but will not implicitly
+    add the `ref` modifier. Instead, the value referred to by the `ref return`
+    is *copied* to the variable on the left-hand side of the assignment. The
+    variable is not a `ref` local.
+
+    In order to get the result you want, you need to add the `ref` modifier
+    to the local variable declaration to make the variable a reference when
+    the return value is a reference:
+
+    [!code-csharp[AssignRefReturn](~/samples/snippets/csharp/new-in-7/program.cs#24_AssignRefReturn "Assign ref return")]
+
+    Now, the second `WriteLine` statement in the example above will print 
+    out the value `24`, indicating that the storage in the matrix has been
+    modified. The local variable has been declared with the `ref` modifier,
+    and it will take a `ref` return. You must initialize a `ref` variable when
+    it is declared, you cannot split the declaration and the initialization.
+
+    The C# language has three other rules that protect you from misusing
+    the `ref` locals and returns:
+
+    * You cannot assign a standard method return value to a `ref` local variable.
+        - That disallows statements like `ref int i = sequence.Count();`
+    * You cannot return a `ref` to a variable whose lifetime does not extend beyond the execution of the method.
+        - That means you cannot return a reference to a local variable or a variable with a similar scope.
+    * `ref` locals and returns can't be used with async methods.
+        - The compiler can't know if the referenced variable has been set to its final value when the async method returns.
+
+    The addition of ref locals and ref returns enable algorithms that are more
+    efficient by avoiding copying values, or performing dereferencing operations
+    multiple times.
+
+    Adding `ref` to the return value is a [source compatible change](../../whats-new/version-update-considerations.md#source-compatible-changes). Existing code compiles, but the ref return value is copied when assigned. Callers must update the storage for the return value to a `ref` local variable to store the return as a reference.
+
+    For more information, see the [ref keyword](../../language-reference/keywords/ref.md) article.
+
+- title: Minimize access to code with local functions
```

Comment:
You might consider moving local functions to earlier. You can then use local functions in some of your ref returns and tuple examples in making them interactive.

Reply:
I kept the same order, but hid the local functions behind the try.net scaffolding.